### PR TITLE
Check attachments for size and type before uploading

### PR DESCRIPTION
**Pull request by drparcon**
This pull request handles the checking of the uploaded attachments to verify if they can be uploaded or not.

This prevents attachments of size larger than 5 MB being uploaded as well as files that are not of the accepted file extensions.

### CODE REVIEW

File: src/components/Attachment.svelte [hunk 1]
```diff
@@ -49,6 +51,14 @@
   function previewFiles(e) {
     error = false;
     function readAndPreview(file) {
+      errorMessage = "";
+
+      if (file.size > 5000000) {
```
Comment (grantyap) [inline src/components/Attachment.svelte:56]: Maybe we can declare a `const` variable somewhere so that this can be easily changed in the future?

**Comment by drparcon** [comment]
Currently these are the accepted file extensions and file types
image
image

Please suggest if there are any that we need to add.

**Comment by HazelArong** [comment]
How about .zip? or .rar?

**Comment by HazelArong** [comment]
Also, wouldn't it raise concern that we are accepting .py file type but not other source code file types like .cpp or .c?

**Comment by drparcon** [comment]
> 
> 
> Also, wouldn't it raise concern that we are accepting .py file type but not other source code file types like .cpp or .c?

That's a problem I encountered. When I tried checking the MIME type sa .py files, text/plain siya so covered na siya dungan sa .txt files under the text/plain MIME type. However, for .cpp and .c files, blank iya MIME type when I tried uploading those files. I don't know why that is. Since the acceptedFileType array does not contain a blank string, dili madawat ang .c and .cpp files and I don't think it is the right idea to just add the "" to the accepted file types.

**Comment by drparcon** [comment]
> 
> 
> How about .zip? or .rar?

Will do.

**Review by grantyap — APPROVED** [review]

**Pull request merged**